### PR TITLE
[YANG MODEL]Sonic Yang support for DTEL

### PR DESCRIPTION
**Pull request by RayWang910012**
#### Why I did it
SONiC Yang support for DTEL
#### How I did it
Added a new sonic-dtel yang file.
#### How to verify it
Added tests for sonic dtel yang.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-dtel.yang [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+                        }
+                    }
+                    default "";
+                    description "DTEL sink ports";
+                }
+
+                leaf int_endpoint {
+                    type string {
+                        pattern "true|false";
+                    }
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-dtel.yang:78]: please use type boolean

File: src/sonic-yang-models/yang-models/sonic-dtel.yang [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+                leaf truncate_size {
+                    type uint32;
+                    description "DTEL report truncate size";
+                }
+
+                leaf udp_dest_port {
+                    type string {
+                        pattern "0|[1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]";
+                    }
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-dtel.yang:142]: Can you use type inet:port-number?

File: src/sonic-yang-models/yang-models/sonic-dtel.yang [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                        }
+                    }
+                }
+
+                leaf qindex {
+                    type string {
+                        pattern "[0-7]?" {
+                            error-message "Invalid queue index";
+                            error-app-tag queue-index-invalid;
+                        }
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-dtel.yang:182]: Is it possible to have more than 8 queues?

File: src/sonic-yang-models/yang-models/sonic-dtel.yang (hunk 1 above)
Comment (RayWang910012) [reply src/sonic-yang-models/yang-models/sonic-dtel.yang]: Got it. Already fixed it.
Please review again.

File: src/sonic-yang-models/yang-models/sonic-dtel.yang (hunk 2 above)
Comment (RayWang910012) [reply src/sonic-yang-models/yang-models/sonic-dtel.yang]: Got it. Already fixed it.
Please review again.

File: src/sonic-yang-models/yang-models/sonic-dtel.yang (hunk 3 above)
Comment (RayWang910012) [reply src/sonic-yang-models/yang-models/sonic-dtel.yang]: It's depended on defferent hardwares.
Change type to uint8.

**Review by ganglyu — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
@RayWang910012 can you please sign the EasyCLA? you can use /easycla to trigger the sign process. Thanks.

**Pull request closed without merging**